internal/config/configgraph_test.go

```diff
@@ -196,3 +202,136 @@ func TestNewConfigGraph(t *testing.T) {
 		})
 	}
 }
+
+func TestConfigGraph_Resolve(t *testing.T) {
+	tests := []struct {
+		name    string
+		URL     string
+		want    Config
+		wantErr bool
+	}{
+		{
+			name: "no includes",
+			URL:  "testdata/include/no_includes.yaml",
+			want: Config{
+				LavaVersion: ptr("v1.0.0"),
+				ChecktypeURLs: []string{
+					"commonchecktypes.json",
+				},
+				Targets: []Target{
+					{
+						Identifier: "example.com",
+						AssetType:  types.DomainName,
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "local file",
+			URL:  "testdata/include/local.yaml",
+			want: Config{
+				Includes:    []string{"testdata/include/no_includes.yaml"},
+				LavaVersion: ptr("v1.0.0"),
+				ChecktypeURLs: []string{
+					"commonchecktypes.json",
+					"checktypes.json",
+				},
+				Targets: []Target{
+					{
+						Identifier: "example.com",
+						AssetType:  types.DomainName,
+					},
+					{
+						Identifier: "example.com",
+						AssetType:  types.DomainName,
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "duplicated",
+			URL:  "testdata/include/duplicated.yaml",
+			want: Config{
+				Includes: []string{
+					"testdata/include/no_includes.yaml",
+					"testdata/include/no_includes.yaml",
+				},
+				LavaVersion: ptr("v1.0.0"),
+				ChecktypeURLs: []string{
+					"commonchecktypes.json",
+					"checktypes.json",
+				},
+				Targets: []Target{
+					{
+						Identifier: "example.com",
+						AssetType:  types.DomainName,
+					},
+					{
+						Identifier: "example.com",
+						AssetType:  types.DomainName,
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "common includes",
+			URL:  "testdata/include/common.yaml",
+			want: Config{
+				Includes: []string{
+					"testdata/include/no_includes.yaml",
+					"testdata/include/no_includes.yaml",
+					"testdata/include/common_a.yaml",
+					"testdata/include/common_b.yaml",
+				},
+				LavaVersion: ptr("v1.0.0"),
+				ChecktypeURLs: []string{
+					"checktypes.json",
+					"commonchecktypes.json",
+					"checktypes.json",
+					"checktypes.json",
+				},
+				Targets: []Target{
+					{
+						Identifier: "example.com",
+						AssetType:  types.DomainName,
+					},
+					{
+						Identifier: "example.com",
+						AssetType:  types.DomainName,
+					},
+					{
+						Identifier: "example.com",
+						AssetType:  types.DomainName,
+					},
+					{
+						Identifier: "example.com",
+						AssetType:  types.DomainName,
+					},
+				},
+				ReportConfig: ReportConfig{
+					Severity: ptr(SeverityCritical),
+				},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			graph, err := NewConfigGraph(tt.URL)
+			if err != nil {
+				t.Errorf("build dag: %v", err)
+			}
+			graph.Resolve()
+			got := graph.cfg
+			if (err != nil) != tt.wantErr {
+				t.Errorf("unexpected error: want: %v, got: %v", tt.wantErr, err)
```

Comment:
Given that `wantErr` is a boolean, I would simplify the error message to:

```suggestion
				t.Errorf("unexpected error: %v", err)
```

If it fails, you know that you wanted the opposite.